p2p/transport/webrtc/udpmux/packetqueue.go

```diff
@@ -0,0 +1,85 @@
+package udpmux
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	pool "github.com/libp2p/go-buffer-pool"
+)
+
+type packet struct {
+	buf []byte
+}
+
+var (
+	errTooManyPackets    = errors.New("too many packets in queue; dropping")
+	errEmptyPacketQueue  = errors.New("packet queue is empty")
+	errPacketQueueClosed = errors.New("packet queue closed")
+)
+
+type packetQueue struct {
+	packetsMux sync.Mutex
+	packetsCh  chan struct{}
+	packets    []packet
+}
+
+func newPacketQueue() *packetQueue {
+	return &packetQueue{
+		packetsCh: make(chan struct{}, maxPacketsInQueue),
```

Comment:
Can we avoid allocating a channel with a capacity of 128 here? That's a lot of memory.
Better solution: Have a signal channel with a cap of 1, and a packet slice (you already have that one). That way, we only consume the memory when we're actually receiving that many packets.